src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -120,15 +123,24 @@ bool RecoveryBehavior(object message)
                         }
                         case LoadSnapshotFailed failed:
                             timeoutCancelable.Cancel();
-                            try
+                            if (snapshotIsOptional)
                             {
-                                OnRecoveryFailure(failed.Cause);
+                                Log.Info("Snapshot load error for persistenceId [{0}]. Replaying all events since snapshot-is-optional=true", PersistenceId);
```

Comment:
LGTM - might consider making this a `WARNING` but I think it's fine as-is for now